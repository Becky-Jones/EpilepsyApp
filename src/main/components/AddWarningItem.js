import React, { useState } from "react";
import { StyleSheet, View, Text, TextInput, Button } from "react-native";
const styles = require("../pages/stylesheets/styles");

export default function addWarningItem(props) {
  const [item, setItem] = useState("");
  return (
    <View>
      <TextInput
        onChangeText={(textVal) => {
          setItem(textVal);
        }}
        style={styles.input}
        placeholder="Warning in format. 1:00 2:00"
        placeholderTextColor="white"
      ></TextInput>
      <Button
        style={thispage.btn}
        onPress={() => {
          props.addWarningItem(item);
        }}
        title="Add warning"
      ></Button>
    </View>
  );
}

const thispage = StyleSheet.create({
  view: {
    width: 350,
    height: 500,
    backgroundColor: "lightgrey",
    margin: 10,
    padding: 8,
    color: "white",
    borderRadius: 14,
    fontSize: 18,
    fontWeight: "500",
  },
  textBox: {
    backgroundColor: "white",
    width: 350,
    height: 30,
    margin: 10,
    padding: 8,
    color: "black",
    borderRadius: 14,
    fontSize: 12,
    fontWeight: "500",
  },
  textInput: {
    color: "white",
    fontSize: 16,
    fontWeight: "500",
  },
  View: {
    width: 350,
    margin: 10,
    backgroundColor: "lightgrey",
    borderRadius: 14,
  },
  btn: {
    width: 200,
    borderRadius: 14,
    color: "red",
  },
});
