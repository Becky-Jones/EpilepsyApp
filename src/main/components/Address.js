export class Address {
  constructor(address, city, postcode) {
    this.address = address;
    this.city = city;
    this.postcode = postcode;
  }

  /*****************************
   * GETTERS
   *****************************/

  getAddressLine() {
    return this.address;
  }

  getCity() {
    return this.city;
  }

  getPostcode() {
    return this.postcode;
  }

  getFullAddress() {
    return this.address + ", " + this.city + ", " + this.postcode;
  }

  /*****************************
   * SETTERS
   *****************************/

  setAddress(address) {
    this.address = address;
  }

  setCity(city) {
    this.city = city;
  }

  setPostcode(postcode) {
    this.postcode = postcode;
  }
}
