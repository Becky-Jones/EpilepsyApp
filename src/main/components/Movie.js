export class Movie {
  constructor(id, title, length, warnings) {
    this.id = id;
    this.title = title;
    this.warnings = warnings;
    this.length = length;
  }
  /*****************************
   * GETTERS
   *****************************/

  getTitle() {
    return this.title;
  }

  getWarnings() {
    return this.warnings.toString();
  }

  getLength() {
    return this.length;
  }

  getId() {
    return this.id;
  }

  /*****************************
   * SETTERS
   *****************************/

  settitle(title) {
    this.title = title;
  }

  setWarnings(warnings) {
    this.warnings = warnings;
  }
  
  setId(id) {
    this.id = id;
  }

  setLength(length) {
    this.length = length;
  }
}
