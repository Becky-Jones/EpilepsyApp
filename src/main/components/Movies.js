export class Movies {
  constructor() {
    this.movies = [];
  }
  /*****************************
   * GETTERS
   *****************************/

  getMovies() {
    return this.movies.toString();
  }

  /*****************************
   * SETTERS
   *****************************/

  setMovies(movies) {
    this.movies = movies;
  }
}
