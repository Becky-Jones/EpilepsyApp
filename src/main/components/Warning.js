export class Warning {
    constructor(
      start_time,
      end_time
    ) {
      this.start_time = start_time;
      this.end_time = end_time;
    }
    /*****************************
     * GETTERS
     *****************************/
  
    getTriggerStart() {
      return this.start_time;
    }
  
    getTriggerEnd() {
      return this.end_time;
    }
  
    /*****************************
     * SETTERS
     *****************************/
  
    setTriggerStart(triggerStart) {
      this.start_time = triggerStart;
    }
  
    setTriggerEnd(triggerEnd) {
      this.end_time = triggerEnd;
    }
  
  
  }
  