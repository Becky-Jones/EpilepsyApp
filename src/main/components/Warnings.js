export class Warnings {
    constructor(
      triggerStart,
      triggerEnd
    ) {
      this.triggerStart = triggerStart;
      this.triggerEnd = triggerEnd;
    }
    /*****************************
     * GETTERS
     *****************************/
  
    getTriggerStart() {
      return this.triggerStart;
    }
  
    getTriggerEnd() {
      return this.triggerEnd;
    }
  
    /*****************************
     * SETTERS
     *****************************/
  
    setTriggerStart(triggerStart) {
      this.triggerStart = triggerStart;
    }
  
    setTriggerEnd(triggerEnd) {
      this.triggerEnd = triggerEnd;
    }
  
  
  }
  