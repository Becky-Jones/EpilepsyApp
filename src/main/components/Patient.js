import { Address } from "./Address";
import { SeizureDetails } from "./seizureDetails";

export class Patient {
  constructor(
    user_id,
    userType,
    firstName,
    surname,
    dob,
    email,
    password,
    gender,
    address,
    city,
    postcode,
    practitionerId,
    seizureTriggers,
    seizureTypes,
    seizureFreq,
    yearsSuffered,
    MHIssues
  ) {
    this.userType = userType;
    this.user_id = user_id;
    this.firstName = firstName;
    this.surname = surname;
    this.dob = dob;
    this.email = email;
    this.password = password;
    this.gender = gender;
    this.Address = new Address(address, city, postcode);
    this.practitionerId = practitionerId;
    this.seizureDetails = new SeizureDetails(
      seizureTriggers,
      seizureTypes,
      seizureFreq,
      yearsSuffered,
      MHIssues
    );
  }
  /*****************************
   * GETTERS
   *****************************/

  getName() {
    return this.firstName + " " + this.surname;
  }

  getFirstName() {
    return this.firstName;
  }

  getSurname() {
    return this.surname;
  }

  getDOB() {
    return this.dob;
  }

  getEmail() {
    return this.email;
  }

  getType() {
    return this.userType;
  }

  getId() {
    return this.user_id;
  }

  getPassword() {
    return this.password;
  }

  getGender() {
    return this.gender;
  }

  getAddress() {
    return this.Address;
  }

  getSeizureDetails() {
    return this.seizureDetails;
  }
  
  getPractitionerId() {
    return this.practitionerId;
  }


  /*****************************
   * SETTERS
   *****************************/

  setFirstName(name) {
    this.firstName = name;
  }

  setSurname(name) {
    this.surname = name;
  }

  setDOB(dob) {
    this.dob = dob;
  }

  setEmail(email) {
    this.email = email;
  }

  setType(type) {
    this.userType = type;
  }

  setPassword(password) {
    this.password = password;
  }

  setGender(gender) {
    this.gender = gender;
  }

  setPractitionerId(id) {
    this.practitionerId = id;
  }
}
