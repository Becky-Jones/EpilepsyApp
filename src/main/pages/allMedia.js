import React, { useState } from "react";
import {
  StyleSheet,
  Text,
  TextInput,
  Button,
  View,
  TouchableOpacity,
  Alert,
} from "react-native";
import { ScrollView } from "react-native-gesture-handler";
import Table from "react-native-simple-table";
import { Movie } from "../components/Movie";
import displayNav from "../components/NavBar";
import { Warning } from "../components/Warning";
import { Warnings } from "../components/Warnings";
import AddWarningItem from "../components/AddWarningItem";
import ListWarningItems from "../components/ListWarningItems";
import { StatusBar } from "expo-status-bar";
import uuid from "react-native-uuid";

const styles = require("./stylesheets/styles");
const mediaDetailsStyle = require("./stylesheets/mediaDetailsStyle");

export default function MediaDetails({ navigation, route }) {
  const [S_AddMediaName, setMediaName] = useState("");
  const [S_MediaLength, setMediaLength] = useState("");
  const [S_TriggerStart, setTriggerStart] = useState("");
  const [S_TriggerEnd, setTriggerEnd] = useState("");
  const [S_Warnings, setWarnings] = useState([]);
  const params = route.params;
  const user = params.User;
  const movies = params.Movies;
  const columnsMediaTable = setupTable();
  const mediaDetailsFormattedForTable = formatMediaDetails(
    movies,
    navigation,
    user,
    movies
  );

  function displayAddMedia(
    S_MediaName,
    setMediaName,
    S_MediaLength,
    setMediaLength
  ) {
    const addWarningItem = (text) => {
      const newWarningItem = {
        warning: text,
        id: uuid.v4()
      };
      setWarningList([newWarningItem, ...warninglist]);
    };
    const DeleteWarningItem = (id) => {
      const newWarningList = warninglist.filter((item) => item.id !== id);
      setWarningList(newWarningList);
    };
    const [warninglist, setWarningList] = useState([]);

    return (
      <>
        <View>
          <TextInput
            placeholder="Enter media name here ... "
            autoCapitalize="none"
            placeholderTextColor="white"
            style={styles.input}
            value={S_MediaName}
            onChangeText={(name) => setMediaName(name)}
          />
          <TextInput
            placeholder="Enter media length here ... "
            autoCapitalize="none"
            placeholderTextColor="white"
            style={styles.input}
            value={S_MediaLength}
            onChangeText={(length) => setMediaLength(length)}
          />
          <View>
            <AddWarningItem addWarningItem={addWarningItem}></AddWarningItem>
            <ListWarningItems
              deleteWarningItem={DeleteWarningItem}
              listWarningItems={warninglist}
            ></ListWarningItems>
            <StatusBar style="auto" />
          </View>
        </View>
        <Button
          title="Add Media to database"
          onPress={() => AddMedia(S_MediaName, S_MediaLength, warninglist)}
        />
      </>
    );
  }

  function AddMedia(mediaName, mediaLength, warninglist) {
    var newMovie = new Movie();
    var warnings = [];

    //setup the list of warnings in the correct format
    for (var x = 0; x < warninglist.length; x++) {
      var warningString = warninglist[x].warning;
      warningString += "";
      var warningArray = warningString.split(" ");
      var warning = new Warning(warningArray[1], warningArray[0]);
      warnings.push(warning);
    }

    //assign all the details to the new film
    newMovie.setTitle(mediaName);
    newMovie.setLength(mediaLength);
    newMovie.setWarnings(warnings);

    fetch("http://192.168.0.7:4000/add-movie", {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        title: newMovie.getTitle(),
        length: newMovie.getLength(),
        warnings: warnings,
      }),
    })
      .then((response) => {
        console.log("Movie added successfully");
        navigation.navigate("All Media");
      })
      .catch((error) => {
        console.log(error);
        alert(
          "Sorry. We couldn't add your movie at this time. Please try again later."
        );
      });
  }

  function RemoveMedia(mediaID) {
    console.log("Removing " + mediaID);
    var url = "http://192.168.0.7:4000/delete-movie/" + mediaID;
    url;
    fetch(url, { method: "DELETE" })
      .then((response) => {
        console.log("Movie removed successfully");
        navigation.navigate("All Media");
      })
      .catch((error) => {
        console.log(error);
        alert(
          "Sorry. We couldn't remove your movie at this time. Please try again later."
        );
      });
  }

  function displayMediaTable(columnsMediaTable, mediaDetailsFormattedForTable) {
    return (
      <View style={mediaDetailsStyle.table}>
        <Table
          height={420}
          columnWidth={150}
          columns={columnsMediaTable}
          dataSource={mediaDetailsFormattedForTable}
        />
      </View>
    );
  }

  //this function fixes a weird problem where array is created within an array
  function formatMediaDetails(dataSourceMediaTable, navigation, user, movies) {
    const mediaDetailsFormattedForTable = [];
    const films = dataSourceMediaTable.movies;
    for (var x = 0; x < films.length; x++) {
      const warning = films[x].warnings;
      const movieId = films[x]._id;
      // console.log(warning);
      mediaDetailsFormattedForTable.push({
        mediaName: films[x].title,
        movieLength: films[x].length,
        linky: (
          <Button
            title="Link"
            onPress={() =>
              navigation.navigate("Media Details", {
                User: user,
                Movies: movies,
                Warnings: warning,
              })
            }
          />
        ),
        remove: <Button title="Remove" onPress={() => RemoveMedia(movieId)} />,
      });
    }
    return mediaDetailsFormattedForTable;
  }

  function setupTable() {
    return [
      {
        title: "Media Name",
        dataIndex: "mediaName",
        width: 80,
      },
      {
        title: "Movie Length",
        dataIndex: "movieLength",
        width: 80,
      },
      {
        title: "Link",
        dataIndex: "linky",
        width: 70,
      },
      {
        title: "Remove Movie",
        dataIndex: "remove",
        width: 90,
      },
    ];
  }

  return (
    <>
      <ScrollView>
        {displayNav(navigation, user, movies)}
        <View style={mediaDetailsStyle.container}>
          <Text style={mediaDetailsStyle.title}>Media Details</Text>
          {displayMediaTable(columnsMediaTable, mediaDetailsFormattedForTable)}
        </View>

        <View style={mediaDetailsStyle.containerBottom}>
          <Text style={mediaDetailsStyle.title}>Add movie</Text>
          <ScrollView>
            {displayAddMedia(
              S_AddMediaName,
              setMediaName,
              S_MediaLength,
              setMediaLength,
              S_TriggerStart,
              setTriggerStart,
              S_TriggerEnd,
              setTriggerEnd,
              S_Warnings,
              setWarnings
            )}
          </ScrollView>
        </View>
      </ScrollView>
    </>
  );
}
