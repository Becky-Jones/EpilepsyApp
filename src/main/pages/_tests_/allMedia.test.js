import React from "react";
import renderer from "react-test-renderer";
import { Patient } from "../../components/Patient";
import { Movies } from "../../components/Movies";
import { Movie } from "../../components/Movie";
import { Warnings } from "../../components/Warnings";
import { Warning } from "../../components/Warning";
import allMedia from '../allMedia';

describe("<allMedia/>", () => {
  it("renders correctly", () => {
    const navigate = jest.fn();
    const user = new Patient(
      "TESTId",
      "Patient",
      "Bob",
      "Bobby",
      "04-08-1998",
      "test@gmail.com",
      "password",
      "male",
      "5 testing lane",
      "liverpool",
      "TR6 0GH",
      "12345678",
      ["Flashing lights"],
      ["Absense Seizure"],
      "4",
      "1",
      ["Anxiety"]
    );
    var moviesX = new Movies();
    var movies =
      '{"movies":[{"_id":"616d9e5ed7e15f4acaf7dbd9","title":"John Wick: Chapter 2","length":"2:2:01","warnings":[{"start_time":"0:40:33","end_time":"0:40:57"},{"start_time":"0:44:35","end_time":"0:44:55"},{"start_time":"1:3:42","end_time":"1:3:59"},{"start_time":"1:16:11","end_time":"1:16:24"},{"start_time":"1:24:35","end_time":"1:24:52"},{"start_time":"1:27:54","end_time":"1:28:12"},{"start_time":"1:34:27","end_time":"1:34:52"},{"start_time":"1:44:10","end_time":"1:44:28"},{"start_time":"1:49:43","end_time":"1:50:03"},{"start_time":"1:57:43","end_time":"1:58:01"}]},{"_id":"61a0ef22452054bb4e975d6a","title":"John Wick","length":"1:41:01","warnings":[{"start_time":"0:13:42","end_time":"0:13:56"},{"start_time":"0:14:16","end_time":"0:14:38"},{"start_time":"0:31:27","end_time":"0:31:41"},{"start_time":"0:50:17","end_time":"0:50:34"},{"start_time":"1:1:15","end_time":"1:1:31"},{"start_time":"1:4:53","end_time":"1:5:12"},{"start_time":"1:11:51","end_time":"1:12:17"},{"start_time":"1:17:53","end_time":"1:18:08"},{"start_time":"1:35:13","end_time":"1:35:34"},{"start_time":"1:39:6","end_time":"1:39:25"},{"start_time":"0:40:33","end_time":"0:40:57"},{"start_time":"0:44:35","end_time":"0:44:55"},{"start_time":"1:3:42","end_time":"1:3:59"},{"start_time":"1:16:11","end_time":"1:16:24"},{"start_time":"1:24:35","end_time":"1:24:52"},{"start_time":"1:27:54","end_time":"1:28:12"},{"start_time":"1:34:27","end_time":"1:34:52"},{"start_time":"1:44:10","end_time":"1:44:28"},{"start_time":"1:49:43","end_time":"1:50:03"},{"start_time":"1:57:43","end_time":"1:58:01"}]},{"_id":"61a242b6e6efa052217ce167","title":"Hitman","length":"1:52:49","warnings":[{"start_time":"0:30:50","end_time":"0:30:04"},{"start_time":"0:02:54","end_time":"0:02:22"},{"start_time":"0:07:04","end_time":"0:06:41"}],"createdAt":"2021-11-27T14:37:42.749Z","updatedAt":"2021-12-01T17:26:38.810Z","__v":0},{"_id":"61a3c3dfb8214d9af1e44910","title":"John Wick: chapter 3","length":"2:20:10","warnings":[{"start_time":"15:20:04","end_time":"14:40:05"},{"start_time":"13:10:44","end_time":"13:40:01"}],"createdAt":"2021-11-28T18:01:03.403Z","updatedAt":"2021-12-03T10:22:24.487Z","__v":0}]}';
    movies = JSON.parse(movies);
    const moviesList = [];
    const moviesArray = movies.movies;
    for (var i = 0; i < moviesArray.length; i++) {
      var warningsList = [];
      const details = moviesArray[i];
      const warningsArray = moviesArray[i].warnings;
      for (var x = 0; x < warningsArray.length; x++) {
        const warning = new Warning(
          warningsArray[x].start_time,
          warningsArray[x].end_time
        );
        warningsList.push(warning);
      }
      const warning = new Warnings(warningsList);
      const movie = new Movie(
        details._id,
        details.title,
        details.length,
        warning
      );
      moviesList.push(movie);
    }
    moviesX.setMovies(moviesList);

    const route = { Patient: "", Movies: moviesList, Patients: "", User: user };
    const tree = renderer
      .create(<allMedia route={route} navigation={navigate} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
