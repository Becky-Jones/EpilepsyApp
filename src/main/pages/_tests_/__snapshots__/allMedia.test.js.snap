// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<allMedia/> renders correctly 1`] = `
<allMedia
  navigation={[MockFunction]}
  route={
    Object {
      "Movies": Array [
        Movie {
          "id": "616d9e5ed7e15f4acaf7dbd9",
          "length": "2:2:01",
          "title": "John Wick: Chapter 2",
          "warnings": Warnings {
            "warningsList": Array [
              Warning {
                "end_time": "0:40:57",
                "start_time": "0:40:33",
              },
              Warning {
                "end_time": "0:44:55",
                "start_time": "0:44:35",
              },
              Warning {
                "end_time": "1:3:59",
                "start_time": "1:3:42",
              },
              Warning {
                "end_time": "1:16:24",
                "start_time": "1:16:11",
              },
              Warning {
                "end_time": "1:24:52",
                "start_time": "1:24:35",
              },
              Warning {
                "end_time": "1:28:12",
                "start_time": "1:27:54",
              },
              Warning {
                "end_time": "1:34:52",
                "start_time": "1:34:27",
              },
              Warning {
                "end_time": "1:44:28",
                "start_time": "1:44:10",
              },
              Warning {
                "end_time": "1:50:03",
                "start_time": "1:49:43",
              },
              Warning {
                "end_time": "1:58:01",
                "start_time": "1:57:43",
              },
            ],
          },
        },
        Movie {
          "id": "61a0ef22452054bb4e975d6a",
          "length": "1:41:01",
          "title": "John Wick",
          "warnings": Warnings {
            "warningsList": Array [
              Warning {
                "end_time": "0:13:56",
                "start_time": "0:13:42",
              },
              Warning {
                "end_time": "0:14:38",
                "start_time": "0:14:16",
              },
              Warning {
                "end_time": "0:31:41",
                "start_time": "0:31:27",
              },
              Warning {
                "end_time": "0:50:34",
                "start_time": "0:50:17",
              },
              Warning {
                "end_time": "1:1:31",
                "start_time": "1:1:15",
              },
              Warning {
                "end_time": "1:5:12",
                "start_time": "1:4:53",
              },
              Warning {
                "end_time": "1:12:17",
                "start_time": "1:11:51",
              },
              Warning {
                "end_time": "1:18:08",
                "start_time": "1:17:53",
              },
              Warning {
                "end_time": "1:35:34",
                "start_time": "1:35:13",
              },
              Warning {
                "end_time": "1:39:25",
                "start_time": "1:39:6",
              },
              Warning {
                "end_time": "0:40:57",
                "start_time": "0:40:33",
              },
              Warning {
                "end_time": "0:44:55",
                "start_time": "0:44:35",
              },
              Warning {
                "end_time": "1:3:59",
                "start_time": "1:3:42",
              },
              Warning {
                "end_time": "1:16:24",
                "start_time": "1:16:11",
              },
              Warning {
                "end_time": "1:24:52",
                "start_time": "1:24:35",
              },
              Warning {
                "end_time": "1:28:12",
                "start_time": "1:27:54",
              },
              Warning {
                "end_time": "1:34:52",
                "start_time": "1:34:27",
              },
              Warning {
                "end_time": "1:44:28",
                "start_time": "1:44:10",
              },
              Warning {
                "end_time": "1:50:03",
                "start_time": "1:49:43",
              },
              Warning {
                "end_time": "1:58:01",
                "start_time": "1:57:43",
              },
            ],
          },
        },
        Movie {
          "id": "61a242b6e6efa052217ce167",
          "length": "1:52:49",
          "title": "Hitman",
          "warnings": Warnings {
            "warningsList": Array [
              Warning {
                "end_time": "0:30:04",
                "start_time": "0:30:50",
              },
              Warning {
                "end_time": "0:02:22",
                "start_time": "0:02:54",
              },
              Warning {
                "end_time": "0:06:41",
                "start_time": "0:07:04",
              },
            ],
          },
        },
        Movie {
          "id": "61a3c3dfb8214d9af1e44910",
          "length": "2:20:10",
          "title": "John Wick: chapter 3",
          "warnings": Warnings {
            "warningsList": Array [
              Warning {
                "end_time": "14:40:05",
                "start_time": "15:20:04",
              },
              Warning {
                "end_time": "13:40:01",
                "start_time": "13:10:44",
              },
            ],
          },
        },
      ],
      "Patient": "",
      "Patients": "",
      "User": Patient {
        "Address": Address {
          "address": "5 testing lane",
          "city": "liverpool",
          "postcode": "TR6 0GH",
        },
        "dob": "04-08-1998",
        "email": "test@gmail.com",
        "firstName": "Bob",
        "gender": "male",
        "password": "password",
        "practitionerId": "12345678",
        "seizureDetails": SeizureDetails {
          "MHIssues": Array [
            "Anxiety",
          ],
          "seizureFreq": "4",
          "seizureTriggers": Array [
            "Flashing lights",
          ],
          "seizureTypes": Array [
            "Absense Seizure",
          ],
          "yearsSuffered": "1",
        },
        "surname": "Bobby",
        "userType": "Patient",
        "user_id": "TESTId",
      },
    }
  }
/>
`;
